include/unified_teleop/controller.hpp

```diff
@@ -0,0 +1,124 @@
+#ifndef TELEOP_CONTROLLER_INCLUDE_GUARD_HPP
+#define TELEOP_CONTROLLER_INCLUDE_GUARD_HPP
+
+#include "rclcpp/rclcpp.hpp"
+#include "sensor_msgs/msg/joy.hpp"
+#include <optional>
+#include <map>
+#include <string>
+#include <cmath>
+#include "yaml-cpp/yaml.h"
+
+namespace rosnu
+{
+    enum class InputType
+    {
+        /// @brief  Represents an axis input, such as a joystick; the values usually start from 0.0 and range from -1.0 to 1.0.
+        Axis,
+        /// @brief Represents a trigger input, such as a trigger button; the values usually start from 1.0 and range from -1.0 to 1.0.
+        Trigger,
+        /// @brief Represents a button input, such as a regular button; the values are usually either 0 or 1.
+        Button,
+        None
+    };
+
+    /// @brief Represents an unused input type.
+    constexpr InputType UNUSED_TYPE = InputType::None;
+
+    /// @brief Represents an unused index value.
+    constexpr int UNUSED_INDEX = -1;
+
+    struct MovementInput
```

Comment:
This should not have public members. It should have private members. There is an invariant here: UNUSED_INDEX is only a valid index when the InputType is None